### ISSUE TITLE
Changes to WYSIWYG don't notify form subscribers

### ISSUE THREAD

**Issue by dwalkr** [tinacms/tinacms#823]
## Update

The problem occurs when you pass an anonymous function into useWatchFormValues. In that case, the subscription is re-attached on every render. Furthermore, since `useWatchFormValues` has logic to skip the callback on the first subscriber event, in this scenario it seems that every subscriber event is treated as if it's the first one, and so the callback is never executed.

It's still a bit puzzling that this appears to specifically affect the WYSIWYG.

Either way, the source of the problem is [this line in use-watch-form-values.ts](https://github.com/tinacms/tinacms/blob/master/packages/%40tinacms/react-core/src/use-watch-form-values.ts#L47). I'm not quite sure of the unintended side effects of removing this watcher and potentially breaking reactivity.

It would be nice to allow anonymous functions to be passed in to `useWatchFormValues`, but if that's not feasible I suppose we should just make sure this pitfall is well communicated in the docs.

## Original Issue

![useWatchFormValues](https://user-images.githubusercontent.com/15221702/75591095-d96c7700-5a4c-11ea-9065-8ee5b5421075.gif)

The above uses `useWatchFormValues` to trigger that "notified" response via something more or less like this:

```javascript
const [updateReceived, setUpdateReceived] = useState(false)
useWatchFormValues(form, () => setUpdateReceived(true))
useEffect(() => {
  if (updateReceived) {
    setTimeout(() => setUpdateReceived(false), 500)
  }
}, [updateReceived])
return (<h1 style={{ color: updateReceived ? 'goldenrod' : 'black' }}>
  {updateReceived ? 'Notified' : 'Waiting'}
</h1>)
```

**Comment by dwalkr** [comment]
A couple notes:

1. `useWatchFormValues` _does_ work with the WYSIWYG in some contexts. The Gatsby and Next.js markdown helpers successfully write to the filesystem when WYSIWYG content is changed, so this may be "higher up in the chain" so to speak.
2. If I replace the call to `useWatchFormValues` with a call to `form.subscribe` everything works as expected

**Comment by dwalkr** [comment]
I think I found the source of the problem. Updated issue.

**Comment by ncphillips** [comment]
This issue makes me think a little Dependency Inversion is in order.  The `deps` array should be passed in by the user:

```ts
const [, form ] = useForm({ ... })

useWatchFormValues(form, values => {
 // do something
}, [form])
```

If they want it to re-render on the callback:

```ts
const [, form ] = useForm({ ... })

const doSomething = values => {
 // do something
}

useWatchFormValues(form, doSomething, [form, doSomething])
```

**Comment by kendallstrautman** [comment]
Closing as this was addressed via #163? Feel free to reopen if not

**Comment by kr2792** [comment]
Did this get fixed? I still have [problems](https://community.tinacms.org/t/usewatchformvalues-doesnt-register-changes-when-typing-in-a-inlinewysiwyg-component/639) with this issue. I've tried to use forms.subscribe to circumvent it, but it's hacky as I compare the values in the formState and the post so see if there are any changes. 

`
	form.subscribe(formState => {
		
		let hasChanged = false
        post.blocks?.filter((block) => block._template === "markdown").map((block, index) => {
          
		const oldBlock = formState.values.blocks.filter((block) => block._template === "markdown")?.[index]
         
		if (oldBlock?.markdownBody !== block.markdownBody) {
            hasChanged = true
          }
        })
        
		setUnsavedChanges(hasChanged)
      },
      { values: true }
    )`
